crates/cxx-qt-gen/src/generator/rust/qenum.rs

```diff
@@ -12,9 +12,17 @@ pub fn generate_cxx_mod_contents(qenums: &[ParsedQEnum]) -> Vec<Item> {
         .iter()
         .flat_map(|qenum| {
             let qenum_item = &qenum.item;
-            let qenum_ident = &qenum.ident;
-            let namespace = &qenum.namespace;
-            let namespace = if namespace.is_empty() {
+            let qenum_ident = &qenum.name.rust_unqualified();
+            let namespace = &qenum.name.namespace();
+
+            // If the QEnum has a namespace inherited from its QObject, it won't have the
```

Comment:
can an enum have a different namespace to it's QObject if it's on a QObject ?

eg should we be blocking that case?

Reply:
Unsure :-D Maybe we need to think what that looks like in C++, and could be solved in another pull request, just something i noticed. Maybe create an issue to discuss it

Reply:
Done in #945